scripts/run_presubmit_checks.py

```diff
@@ -52,24 +55,28 @@
     help='optional; if specified, the origin branch to compare against.')
 
 
-def main(args=None) -> None:
+def main(args: Optional[List[str]] = None) -> None:
     """Run the presubmit checks."""
     parsed_args = _PARSER.parse_args(args=args)
 
     # Run Javascript and Python linters.
     print('Linting files since the last commit')
-    pre_commit_linter.main(args=[])
+    pre_commit_linter.main(args=[])  # type: ignore[no-untyped-call]
     print('Linting passed.')
     print('')
 
-    current_branch = subprocess.check_output([
-        'git', 'rev-parse', '--abbrev-ref', 'HEAD'])
+    current_branch = subprocess.check_output(
+        ['git', 'rev-parse', '--abbrev-ref', 'HEAD'], text=True
+    )
 
     # If the current branch exists on remote origin, matched_branch_num=1
     # else matched_branch_num=0.
-    matched_branch_num = subprocess.check_output([
-        'git', 'ls-remote', '--heads', 'origin', current_branch, '|', 'wc',
-        '-l'])
+    matched_branch_num = subprocess.check_output(
+        [
+        'git', 'ls-remote', '--heads', 'origin', current_branch, '|', 'wc', '-l'
+        ],
+        text=True
```

Comment:
Why are we adding this argument in various places?

Reply:
The `subprocess.check_output` method is returning a string value at runtime but while type checking its return type is bytes because previously while calling this method we are not matching any overload so its return type is bytes but after defining the text argument as True. we matched the overload that returns a string type. Reference: https://github.com/python/typeshed/blob/fe3a34503cb54f30c932c329d40e615be14c1ddf/stdlib/subprocess.pyi#L1689

Reply:
Also, the indentation is fixed!

Reply:
Sorry, I don't understand your explanation. By default I believe text=None. Wouldn't that match [this overload](https://github.com/python/typeshed/blob/fe3a34503cb54f30c932c329d40e615be14c1ddf/stdlib/subprocess.pyi#L1692) and return a str as well?

Reply:
@seanlip Just checked out why it is coming as `bytes`. The overloaded function you pointed out requires `encoding` kwarg as `str` rather than `str | None`

Took the signature diff of only a few overload functions tho.

I guess we might have to pass `text` kwarg. This is the playground [URL](https://mypy-play.net/?mypy=latest&python=3.7&gist=bc653d54f065f708143b616b2ff09471) in which I checked the return type.

Reply:
Sorry, still don't understand. Why is the behaviour during runtime and typechecking different, in terms of the returned type? That doesn't make sense to me -- what's different in the two envs that causes this?

Reply:
As per the offline conversation, encoding arg is added wherever possible!